go-controller/pkg/node/gateway_localnet.go

```diff
@@ -116,32 +122,107 @@ func generateGatewayNATRules(ifname string, ip net.IP) []iptRule {
 	return rules
 }
 
-func localnetGatewayNAT(ipt util.IPTablesHelper, ifname string, ip net.IP) error {
-	rules := generateGatewayNATRules(ifname, ip)
-	return addIptRules(ipt, rules)
+func localnetGatewayNAT(localNetdata []*localnetData, ifname string) error {
+	for _, lnData := range localNetdata {
+		rules := generateGatewayNATRules(ifname, lnData.gatewayIP)
+		err := addIptRules(lnData.ipt, rules)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func initLocalnetGateway(nodeName string, subnet *net.IPNet, wf *factory.WatchFactory, nodeAnnotator kube.Annotator) error {
-	// Create a localnet OVS bridge.
+func initLocalnetGateway(nodeName string, subnets []*net.IPNet, wf *factory.WatchFactory, nodeAnnotator kube.Annotator) error {
+
 	localnetBridgeName := "br-local"
-	_, stderr, err := util.RunOVSVsctl("--may-exist", "add-br",
-		localnetBridgeName)
+
+	// Evaluate input subnets
+	err := evaluateInputNetworks(subnets)
```

Comment:
This comment doesn't provide any information beyond what the function name says. It should be removed.

But also, you don't need to validate the data anyway. That gets done by higher levels. You can assume that the data that was passed to you here is a supported configuration.

Reply:
Will remove this method and tests.